src/utils/jsx.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import * as t from "babel-types";
+import type {
+  BabelNodeExpression,
+  BabelNodeArrayExpression,
+  BabelNodeJSXElement,
+  BabelNodeJSXMemberExpression,
+  BabelNodeJSXIdentifier,
+  BabelNodeIdentifier,
+  BabelNodeMemberExpression,
+} from "babel-types";
+import invariant from "../invariant.js";
+import { Value, ObjectValue, SymbolValue, StringValue } from "../values/index.js";
+import { Get } from "../methods/index.js";
+
+export function isReactElement(val: Value): boolean {
+  if (val instanceof ObjectValue && val.properties.has("$$typeof")) {
+    let realm = val.$Realm;
+    let $$typeof = Get(realm, val, "$$typeof");
+    if ($$typeof instanceof SymbolValue) {
+      let existsInGlobalRegistry = realm.globalSymbolRegistry.find(e => e.$Symbol === $$typeof) !== undefined;
+      let description = $$typeof.$Description;
+      return existsInGlobalRegistry && description instanceof StringValue && description.value === "react.element";
```

Comment:
This is actually incorrect. The `$Description` of the symbol is basically the `toString` but not necessarily the global key.

E.g. this is false:
```js
Symbol('react.element') === Symbol.for('react.element') // false
```

But they both have the same description.

You want to check `e.$Key` just like `Symbol.for` does. https://github.com/facebook/prepack/blob/521d5df69c553b71b266cbd20ab035fd41047d10/src/intrinsics/ecma262/Symbol.js#L47-L52

Reply:
Thanks, I'll work on a fix for this once I wake up. :)